src/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureSVN.java

```diff
@@ -220,6 +222,22 @@ public int getWascId() {
 	 */
 	private boolean findSourceCodeSVN(HttpMessage originalMessage) throws Exception {
 
+		//if we got a 404 or a redirect specifically, then this is NOT a match
+		//note that since we are simply relying on the file existing or not, we 
+		//will not attempt any fuzzy matching. Old school.
+		//Checks based on this are necessary, otherwise a recursive scan on nodes 
+		//in the url path cause lots of false positives.
+		List<HttpStatusCode> unwantedCodes= new ArrayList<HttpStatusCode>();
```

Comment:
The type argument should be Integer not HttpStatusCode (it's the former that will be checked later on).
I guess this could be a constant? Avoids recreating the list all the time.